apps/finance/app/src/script.js

```diff
@@ -115,43 +115,41 @@ async function initialize(vaultAddress, ethAddress) {
     async (state, event) => {
       const { vault } = settings
       const { address: eventAddress, event: eventName } = event
-      let nextState = {
+      const nextState = {
         ...state,
       }
 
+      if (event === events.SYNC_STATUS_SYNCING) {
+        return { ...nextState, isSyncing: true }
+      } else if (event === events.SYNC_STATUS_SYNCED) {
+        return { ...nextState, isSyncing: false }
+      }
+
+      // Vault event
       if (addressesEqual(eventAddress, vault.address)) {
-        // Vault event
-        nextState = await vaultLoadBalance(nextState, event, settings)
-      } else {
-        // Finance event
-        switch (eventName) {
-          case events.SYNC_STATUS_SYNCING:
-            nextState.isSyncing = true
-            break
-          case events.SYNC_STATUS_SYNCED:
-            nextState.isSyncing = false
-            break
-          case 'ChangePeriodDuration':
-            nextState.periodDuration = marshallDate(
-              event.returnValues.newDuration
-            )
-            break
-          case 'NewPeriod':
+        return vaultLoadBalance(nextState, event, settings)
+      }
+
+      // Finance event
+      switch (eventName) {
+        case 'ChangePeriodDuration':
+          nextState.periodDuration = marshallDate(
+            event.returnValues.newDuration
+          )
+          return nextState
+        case 'NewPeriod':
+          return {
+            ...(await newPeriod(nextState, event, settings)),
```

Comment:
Spread new state here too:

```suggestion
            ...newState
            ...(await newPeriod(nextState, event, settings)),
```